src/gui_common/CustomDropDown.cs

```diff
@@ -0,0 +1,179 @@
+            Color = color,
+            Id = id,
+            Checkable = checkable,
+        };
+
+        items.Add(item);
+
+        if (checkable)
+        {
+            Popup.AddCheckItem(item.Text, item.Id);
+        }
+        else
+        {
+            Popup.AddItem(item.Text, id);
+        }
+
+        // Redraw the menu button and popup
+        Popup.Update();
+        Update();
+    }
+
+    private void RedrawPopup()
+    {
+        ReadjustRectSizes();
+        DrawIcons();
+    }
+
+    /// <summary>
+    ///   This readjust the rect size of this MenuButton and its PopupMenu.
+    ///   Called when they are to be redrawn.
+    /// </summary>
+    private void ReadjustRectSizes()
+    {
+        // Set popup to minimum length
+        Popup.RectSize = new Vector2(GetContentsMinimumSize().x + iconSize.x + 6, 0);
+
+        // Adjust the menu button to have the same length as the popup
+        RectMinSize = new Vector2(Popup.RectSize.x, RectMinSize.y);
+    }
+
+    /// <summary>
+    ///   A workaround to get PopupMenu icons drawn in an adjusted size.
+    /// </summary>
+    private void DrawIcons()
+    {
+        if (!Popup.Visible)
+            return;
+
+        var contentMinSize = GetContentsMinimumSize();
+        var font = Popup.GetFont("font");
+
+        // Offset from the top
+        var height = Popup.GetStylebox("panel").ContentMarginTop + (font.GetHeight() / 2) - (iconSize.y / 2);
+
+        foreach (var item in items)
+        {
+            // Skip if item has no icon
+            if (item.Icon == null)
+                continue;
+
+            var position = new Vector2(contentMinSize.x, height);
+
+            Popup.DrawTextureRect(item.Icon, new Rect2(position, iconSize), false, item.Color);
+
+            height += font.GetHeight() + cachedPopupVSeparation;
+        }
+    }
+
+    /// <summary>
+    ///   Returns the size of the popup menu contents
+    /// </summary>
+    private Vector2 GetContentsMinimumSize()
+    {
+        var minWidth = 0.0f;
+        var minHeight = 0.0f;
+
+        var font = Popup.GetFont("font");
+
+        foreach (var item in items)
+        {
+            var width = iconSize.x + font.GetStringSize(item.Text).x + cachedPopupHSeparation;
+            var height = Mathf.Max(iconSize.y, font.GetHeight()) + cachedPopupVSeparation;
+
+            if (item.Checkable)
+                width += Popup.GetIcon("checked").GetWidth();
+
+            minWidth = Mathf.Max(minWidth, width);
+            minHeight += height;
+        }
+
+        return new Vector2(minWidth, minHeight);
+    }
+
+    private void OnPopupAboutToShow()
+    {
+        Popup.AddConstantOverride("vseparation", -14);
+
+        // Animate slide down
+        tween.InterpolateProperty(Popup, "custom_constants/vseparation", -14, cachedPopupVSeparation, 0.1f,
+            Tween.TransitionType.Circ, Tween.EaseType.Out);
+        tween.Start();
+    }
+
+    /// <summary>
+    ///   Helper data regarding the item
```

Comment:
`the item` doesn't really tell me what's going on. It should be specified what "item" is here.